### PR TITLE
LWDEV-9731 Turn on maintenance mode when rabbit mq connections are lost

### PR DESCRIPTION
**Pull request by rocketmannot**

### CODE REVIEW

File: src/main/kotlin/com/lykke/matching/engine/services/MessageSender.kt [hunk 1]
```diff
@@ -7,7 +7,7 @@ import java.util.concurrent.BlockingQueue
 
 @Component
 class MessageSender(private val clientsEventsQueue: BlockingQueue<Event<*>>,
-                    private val trustedClientsEventsQueue: BlockingQueue<ExecutionEvent>) {
+                    private val trustedClientsEventsQueue: BlockingQueue<Event<*>>) {
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/matching/engine/services/MessageSender.kt]: Why is this changed? only ExecutionEvent can be trusted client event

File: src/main/kotlin/com/lykke/matching/engine/outgoing/rabbit/impl/publishers/AbstractRabbitMqPublisher.kt [hunk 2]
```diff
@@ -10,19 +12,23 @@ import com.rabbitmq.client.Channel
...
-                                               private val exchangeType: BuiltinExchangeType,
-                                               private val LOGGER: ThrottlingLogger,
-                                               private val MESSAGES_LOGGER: Logger,
-                                               private val METRICS_LOGGER: MetricsLogger,
-                                               private val STATS_LOGGER: Logger,
-                                               /** null if do not need to log */
+abstract class AbstractRabbitMqPublisher<T>(private val uri: String,
+                                            private val exchangeName: String,
+                                            private val queueName: String,
+                                            val queue: BlockingQueue<out T>,
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/outgoing/rabbit/impl/publishers/AbstractRabbitMqPublisher.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/services/MessageSender.kt (hunk 1 above)
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/services/MessageSender.kt]: fixed
